fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/domain/SavingsAccountTransaction.java

```diff
@@ -149,6 +149,46 @@ public final class SavingsAccountTransaction extends AbstractPersistableCustom {
         this.createdDate = null;
     }
 
+    private SavingsAccountTransaction(final SavingsAccount savingsAccount, final Office office, final PaymentDetail paymentDetail,
+            final Integer typeOf, final LocalDate transactionLocalDate, final LocalDateTime createdDate, final BigDecimal amount,
+            final boolean isReversed, final AppUser appUser, final boolean isManualTransaction, final Boolean lienTransaction,
+            final String refNo) {
+        initializeCommonFields(typeOf, transactionLocalDate, createdDate, amount, isReversed, isManualTransaction, lienTransaction, refNo);
+
+        this.savingsAccount = savingsAccount;
+        this.office = office;
+        this.paymentDetail = paymentDetail;
+        this.appUser = appUser;
+    }
+
+    private SavingsAccountTransaction(final Integer transactionTypeEnum, final LocalDate transactionDate, final BigDecimal amount,
```

Comment:
AFAIS this constructor is only called from one static initialiser 'from' method, which is not used, so probably you can remove that method and so this constructor, and then no need for initializeCommonFields. What do you think?